packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java

```diff
@@ -122,6 +122,48 @@ void init() {
 		taskService.addResponseHandler(equationsCleanupResponseHandler);
 	}
 
+	/**
+	 * Clean up a list of equations
+	 *
+	 * @return List of cleaned-up equations
+	 */
+	@PostMapping("/get-cleaned-equations")
+	@Secured(Roles.USER)
+	public ResponseEntity<List<String>> getCleanedEquations(
+		@RequestBody final List<String> equations,
+		@RequestParam(name = "project-id", required = false) final UUID projectId
+	) {
+		TaskRequest cleanupReq = cleanupEquationsTaskRequest(projectId, equations);
+		TaskResponse cleanupResp = null;
+		try {
+			cleanupResp = taskService.runTask(TaskMode.SYNC, cleanupReq);
+		} catch (final JsonProcessingException e) {
+			log.warn("Unable to clean-up equations due to a JsonProcessingException. Reverting to original equations.", e);
+		} catch (final TimeoutException e) {
+			log.warn("Unable to clean-up equations due to a TimeoutException. Reverting to original equations.", e);
+		} catch (final InterruptedException e) {
+			log.warn("Unable to clean-up equations due to a InterruptedException. Reverting to original equations.", e);
+		} catch (final ExecutionException e) {
+			log.warn("Unable to clean-up equations due to a ExecutionException. Reverting to original equations.", e);
```

Comment:
These `Reverting to original equations` logging doesn't make sense, I don't see any reverting process and you are just returning empty array if some exception happens. Are we actually trying to do something?

Reply:
Just added a proper revert to original equations by adding a fallback